tests/src/CLRTest.Execute.Batch.targets

```diff
@@ -72,20 +72,23 @@ WARNING:   When setting properties based on their current state (for example:
 $(BatchCLRTestEnvironmentCompatibilityCheck)
 IF NOT "%COMPlus_GCStress%"=="" (
   ECHO SKIPPING EXECUTION BECAUSE COMPlus_GCStress IS SET
+  popd
```

Comment:
Is [this](https://github.com/dotnet/coreclr/blob/36e988ee6593351f1e962118a99154df5eae0254/tests/src/CLRTest.Execute.Batch.targets#L347) the corresponding pushd?  I could have sworn the setlocal would preserve the working dir on exit, but if you're adding this I assume you've found that it doesn't.  If you're going to fix these, could you instead define exit blocks (possibly in _CLRTestExecutionScriptText) like:

Exit_success:
    popd
    Exit /b 0

Exit_1
    popd
    Exit /b 1

...

And then replace all the existing Exits with gotos?